ios/Scripts/upload_dsym_manual.sh

```diff
@@ -41,58 +74,132 @@ if [ "${#DSYM_TGZ_FILES[@]}" -eq 0 ]; then
+
+echo ""
+echo "Uploading Build Metadata..."
+RESPONSE_BODY_FILE=$(mktemp)
+TEMP_FILES+=("$RESPONSE_BODY_FILE")
+
+HTTP_STATUS_CODE=$(eval "$CURL_COMMAND_META --write-out '%{http_code}' --silent --output $RESPONSE_BODY_FILE")
+HTTP_RESPONSE_BODY=$(cat "$RESPONSE_BODY_FILE")
+
+if [[ "$HTTP_STATUS_CODE" -ne 200 && "$HTTP_STATUS_CODE" -ne 201 ]]; then
+    echo "[ERROR]: Metadata upload failed with status code $HTTP_STATUS_CODE."
+    exit 1
+fi
+
+echo ""
+echo "Uploading dSYM files..."
+UPLOAD_SUCCESS=true
+MAX_ATTEMPTS=3
+
+echo "$HTTP_RESPONSE_BODY" | jq -c '.mappings[]' | \
+while IFS= read -r URL_OBJECT; do
+    
+    SIGNED_URL=$(echo "$URL_OBJECT" | jq -r '.upload_url')
+    EXPECTED_FILENAME=$(echo "$URL_OBJECT" | jq -r '.filename')
+    
+    UPLOAD_HEADERS_CURL=""
+    HEADER_LINES=$(echo "$URL_OBJECT" | jq -r '.headers | to_entries[] | "\(.key): \(.value)"')
+
+    while IFS= read -r HEADER_LINE; do
+        if [ -n "$HEADER_LINE" ]; then
+            UPLOAD_HEADERS_CURL+=" --header \"$HEADER_LINE\""
+        fi
+    done <<< "$HEADER_LINES"
+    
+    if [ -z "$SIGNED_URL" ] || [ -z "$EXPECTED_FILENAME" ]; then
+        echo "[ERROR]: Failed to extract SIGNED_URL or FILENAME from JSON object. Aborting remaining uploads."
+        UPLOAD_SUCCESS=false
+        break
+    fi
+    
+    DSYM_TGZ_PATH=""
+    for (( j=0; j<${#DSYM_TGZ_BASENAMES[@]}; j++ )); do
+        if [ "${DSYM_TGZ_BASENAMES[$j]}" == "$EXPECTED_FILENAME" ]; then
+            DSYM_TGZ_PATH="${DSYM_TGZ_FILES[$j]}"
+            break
+        fi
+    done
+
+    if [ -z "$DSYM_TGZ_PATH" ]; then
+        echo "[WARNING]: Local dSYM file not found for filename $EXPECTED_FILENAME. Skipping upload."
+        continue
+    fi
+    
+    UPLOAD_ATTEMPT_SUCCESS=false
+    FILE_UPLOAD_COMMAND="curl --request PUT \
+        --url \"$SIGNED_URL\" \
+        --header 'Content-Type: application/octet-stream' \
+        $UPLOAD_HEADERS_CURL \
+        --write-out '%{http_code}' \
+        --silent \
+        --output /dev/null \
+        --data-binary \"@$DSYM_TGZ_PATH\""
+
+    for ATTEMPT in $(seq 1 $MAX_ATTEMPTS); do
+        echo "  Attempt $ATTEMPT/$MAX_ATTEMPTS: Uploading $EXPECTED_FILENAME..."
+        
+        FILE_UPLOAD_STATUS=$(eval "$FILE_UPLOAD_COMMAND")
+        
+        if [[ "$FILE_UPLOAD_STATUS" -ge 200 && "$FILE_UPLOAD_STATUS" -le 299 ]]; then
+            echo "  [SUCCESS]: $EXPECTED_FILENAME uploaded on attempt $ATTEMPT. Status: $FILE_UPLOAD_STATUS"
+            UPLOAD_ATTEMPT_SUCCESS=true
+            break # Success, move to the next file
+        else
+            if [ "$ATTEMPT" -lt "$MAX_ATTEMPTS" ]; then
+                echo "  [WARNING]: $EXPECTED_FILENAME upload failed (Status: $FILE_UPLOAD_STATUS). Retrying in 5 seconds..."
+                sleep 5
+            else
+                echo "  [ERROR]: $EXPECTED_FILENAME upload failed after $MAX_ATTEMPTS attempts. Final Status: $FILE_UPLOAD_STATUS"
```

Comment:
`[ERROR]: Failed to upload ($EXPECTED_FILENAME) after $MAX_ATTEMPTS attempts with status code: $FILE_UPLOAD_STATUS`. Stack traces will not be symbolicated.`